quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java

```diff
@@ -0,0 +1,103 @@
+package com.upgrad.quora.api.controller;
+
+import com.upgrad.quora.api.model.QuestionDetailsResponse;
+import com.upgrad.quora.service.business.AuthenticationService;
+import com.upgrad.quora.service.business.QuestionsService;
+import com.upgrad.quora.service.dao.UserDao;
+import com.upgrad.quora.service.entity.QuestionEntity;
+import com.upgrad.quora.service.entity.UserAuthTokenEntity;
+import com.upgrad.quora.service.entity.UserEntity;
+import com.upgrad.quora.service.exception.AuthenticationFailedException;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/")
+public class QuestionController {
+
+    @Autowired
+    private QuestionsService questionsService;
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @Autowired
```

Comment:
Why do you need Dao object in the Controller and not the business logic

Reply:
Oh. While testing added this. This is not required. Removed the DAO Object.